source/common/http/conn_manager_impl.cc

```diff
@@ -1282,6 +1282,29 @@ void ConnectionManagerImpl::ActiveStream::refreshCachedRoute() {
         connection_manager_.cluster_manager_.get(stream_info_.route_entry_->clusterName());
     cached_cluster_info_ = (nullptr == local_cluster) ? nullptr : local_cluster->info();
   }
+  refreshCachedTracingCustomTags();
+}
+
+void ConnectionManagerImpl::ActiveStream::refreshCachedTracingCustomTags() {
+  if (!hasCachedRoute() || !cached_route_.value()->tracingConfig()) {
+    tracing_custom_tags_.clear();
+    return;
+  }
+  const std::vector<Tracing::CustomTag>& route_tags =
+      cached_route_.value()->tracingConfig()->getCustomTags();
+  if (route_tags.empty()) {
+    tracing_custom_tags_.clear();
+    return;
+  }
+  auto cmp = [](const Tracing::CustomTag& a, const Tracing::CustomTag& b) { return a.tag < b.tag; };
+  std::set<Tracing::CustomTag, decltype(cmp)> merged_tags(route_tags.begin(), route_tags.end(),
```

Comment:
i think this means that you can get multiple `CustomTag`s for the same tag, is that intentional? If not, perhaps you want to use a map of key -> CustomTag instead of a set?

Reply:
use map instead